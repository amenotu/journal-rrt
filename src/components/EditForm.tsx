import * as React from 'react';
import { Button, Modal, Form, Input, TextArea } from 'semantic-ui-react';


export interface Props {
    open: Boolean;
    currentDate: string;
    currentText: string;
    currentIndex: number;
    onEditEntry: (date: string, text: string, index: number) => void;
}

export interface State {
    open: Boolean;
    closeOnEscape?: Boolean;
    closeOnDimmerClick?: Boolean;
    currentDate: string;
    currentText: string;
}


class ModalExampleCloseConfig extends React.Component<Props, State> {
  state = { open: false, closeOnEscape: undefined, closeOnDimmerClick: undefined, currentDate: this.props.currentDate, currentText: this.props.currentText }

  closeConfigShow = (closeOnEscape: Boolean, closeOnDimmerClick: Boolean) => () => {
    this.setState({ closeOnEscape, closeOnDimmerClick, open: true })
  }

  close = () => this.setState({ open: false })

  handleDateChange = (e: React.ChangeEvent<HTMLInputElement>) => this.setState({ currentDate: e.target.value })

  handleTextChange = (e: React.ChangeEvent<HTMLInputElement>) => this.setState({ currentText: e.target.value })

  handleEditSubmit = () => {
    const { currentDate, currentText, } = this.state;
    const {  onEditEntry, currentIndex } = this.props;

    onEditEntry(currentDate, currentText, currentIndex);
    this.setState({open: false})
  }

  render() {
    const { open, closeOnEscape, closeOnDimmerClick, currentDate, currentText } = this.state
    const {currentIndex} = this.props

    return (
      <div key= { currentIndex }>
        <Button onClick={this.closeConfigShow(true, true)}>Edit Entry</Button>

        <Modal
          open={open}
          closeOnEscape={closeOnEscape}
          closeOnDimmerClick={closeOnDimmerClick}
          onClose={this.close}
        >
          <Modal.Header>Edit Your Entry</Modal.Header>
          <Modal.Content>
            <p>Edit your entry below</p>
            <Form onSubmit={ this.handleEditSubmit }>
                <Form.Group>
                    <Form.Field 
                        control= { Input }
                        label='Date'
                        placeholder='Enter the date here...'
                        onChange={ (e: React.ChangeEvent<HTMLInputElement>) => this.handleDateChange(e) } 
                        name='date' 
                        value={ currentDate }
                        required/>
                    <Form.Field 
                        control= { TextArea } 
                        label='Entry' 
                        placeholder='Start typing your entry here...'
                        name='text'
                        value={ currentText}
                        onChange={ (e: React.ChangeEvent<HTMLInputElement>) => this.handleTextChange(e) }
                        required/>
                    <Form.Button content='Submit' />
                </Form.Group>
            </Form>
          </Modal.Content>
          <Modal.Actions>
            <Button onClick={this.close} negative>
              Cancel
            </Button>
          </Modal.Actions>
        </Modal>
      </div>
    )
  }
}

export default ModalExampleCloseConfig;
