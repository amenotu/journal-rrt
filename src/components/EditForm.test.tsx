import * as React from 'react';
import { configure } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import EditForm from './EditForm';

configure({ adapter: new EnzymeAdapter() });

test('EditForm renders', () => {
  const editform = shallow(<EditForm open={true} currentDate='string' currentText='string' currentIndex={0} onEditEntry={ () => {} }/>);
  
  expect(editform).toMatchSnapshot();
});

