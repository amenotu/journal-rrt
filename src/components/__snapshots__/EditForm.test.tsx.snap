// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditForm renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalExampleCloseConfig
    currentDate="string"
    currentIndex={0}
    currentText="string"
    onEditEntry={[Function]}
    open={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          as="button"
          onClick={[Function]}
        >
          Edit Entry
        </Button>,
        <Modal
          centered={true}
          closeOnDimmerClick={true}
          closeOnDocumentClick={false}
          dimmer={true}
          eventPool="Modal"
          onClose={[Function]}
          open={false}
        >
          <ModalHeader>
            Edit Your Entry
          </ModalHeader>
          <ModalContent>
            <p>
              Edit your entry below
            </p>
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormGroup>
                <FormField
                  control={[Function]}
                  label="Date"
                  name="date"
                  onChange={[Function]}
                  placeholder="Enter the date here..."
                  required={true}
                  value="string"
                />
                <FormField
                  control={[Function]}
                  label="Entry"
                  name="text"
                  onChange={[Function]}
                  placeholder="Start typing your entry here..."
                  required={true}
                  value="string"
                />
              </FormGroup>
            </Form>
          </ModalContent>
          <ModalActions>
            <Button
              as="button"
              negative={true}
              onClick={[Function]}
            >
              Cancel
            </Button>
            <Button
              as="button"
              content="Submit"
              icon="checkmark"
              labelPosition="right"
              onClick={[Function]}
              positive={true}
            />
          </ModalActions>
        </Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "Edit Entry",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit Entry",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "centered": true,
          "children": Array [
            <ModalHeader>
              Edit Your Entry
            </ModalHeader>,
            <ModalContent>
              <p>
                Edit your entry below
              </p>
              <Form
                as="form"
                onSubmit={[Function]}
              >
                <FormGroup>
                  <FormField
                    control={[Function]}
                    label="Date"
                    name="date"
                    onChange={[Function]}
                    placeholder="Enter the date here..."
                    required={true}
                    value="string"
                  />
                  <FormField
                    control={[Function]}
                    label="Entry"
                    name="text"
                    onChange={[Function]}
                    placeholder="Start typing your entry here..."
                    required={true}
                    value="string"
                  />
                </FormGroup>
              </Form>
            </ModalContent>,
            <ModalActions>
              <Button
                as="button"
                negative={true}
                onClick={[Function]}
              >
                Cancel
              </Button>
              <Button
                as="button"
                content="Submit"
                icon="checkmark"
                labelPosition="right"
                onClick={[Function]}
                positive={true}
              />
            </ModalActions>,
          ],
          "closeOnDimmerClick": true,
          "closeOnDocumentClick": false,
          "closeOnEscape": undefined,
          "dimmer": true,
          "eventPool": "Modal",
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Edit Your Entry",
            },
            "ref": null,
            "rendered": "Edit Your Entry",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Edit your entry below
                </p>,
                <Form
                  as="form"
                  onSubmit={[Function]}
                >
                  <FormGroup>
                    <FormField
                      control={[Function]}
                      label="Date"
                      name="date"
                      onChange={[Function]}
                      placeholder="Enter the date here..."
                      required={true}
                      value="string"
                    />
                    <FormField
                      control={[Function]}
                      label="Entry"
                      name="text"
                      onChange={[Function]}
                      placeholder="Start typing your entry here..."
                      required={true}
                      value="string"
                    />
                  </FormGroup>
                </Form>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit your entry below",
                },
                "ref": null,
                "rendered": "Edit your entry below",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "form",
                  "children": <FormGroup>
                    <FormField
                      control={[Function]}
                      label="Date"
                      name="date"
                      onChange={[Function]}
                      placeholder="Enter the date here..."
                      required={true}
                      value="string"
                    />
                    <FormField
                      control={[Function]}
                      label="Entry"
                      name="text"
                      onChange={[Function]}
                      placeholder="Start typing your entry here..."
                      required={true}
                      value="string"
                    />
                  </FormGroup>,
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <FormField
                        control={[Function]}
                        label="Date"
                        name="date"
                        onChange={[Function]}
                        placeholder="Enter the date here..."
                        required={true}
                        value="string"
                      />,
                      <FormField
                        control={[Function]}
                        label="Entry"
                        name="text"
                        onChange={[Function]}
                        placeholder="Start typing your entry here..."
                        required={true}
                        value="string"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "control": [Function],
                        "label": "Date",
                        "name": "date",
                        "onChange": [Function],
                        "placeholder": "Enter the date here...",
                        "required": true,
                        "value": "string",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "control": [Function],
                        "label": "Entry",
                        "name": "text",
                        "onChange": [Function],
                        "placeholder": "Start typing your entry here...",
                        "required": true,
                        "value": "string",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Button
                  as="button"
                  negative={true}
                  onClick={[Function]}
                >
                  Cancel
                </Button>,
                <Button
                  as="button"
                  content="Submit"
                  icon="checkmark"
                  labelPosition="right"
                  onClick={[Function]}
                  positive={true}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": "Cancel",
                  "negative": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "content": "Submit",
                  "icon": "checkmark",
                  "labelPosition": "right",
                  "onClick": [Function],
                  "positive": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            as="button"
            onClick={[Function]}
          >
            Edit Entry
          </Button>,
          <Modal
            centered={true}
            closeOnDimmerClick={true}
            closeOnDocumentClick={false}
            dimmer={true}
            eventPool="Modal"
            onClose={[Function]}
            open={false}
          >
            <ModalHeader>
              Edit Your Entry
            </ModalHeader>
            <ModalContent>
              <p>
                Edit your entry below
              </p>
              <Form
                as="form"
                onSubmit={[Function]}
              >
                <FormGroup>
                  <FormField
                    control={[Function]}
                    label="Date"
                    name="date"
                    onChange={[Function]}
                    placeholder="Enter the date here..."
                    required={true}
                    value="string"
                  />
                  <FormField
                    control={[Function]}
                    label="Entry"
                    name="text"
                    onChange={[Function]}
                    placeholder="Start typing your entry here..."
                    required={true}
                    value="string"
                  />
                </FormGroup>
              </Form>
            </ModalContent>
            <ModalActions>
              <Button
                as="button"
                negative={true}
                onClick={[Function]}
              >
                Cancel
              </Button>
              <Button
                as="button"
                content="Submit"
                icon="checkmark"
                labelPosition="right"
                onClick={[Function]}
                positive={true}
              />
            </ModalActions>
          </Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Edit Entry",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit Entry",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "centered": true,
            "children": Array [
              <ModalHeader>
                Edit Your Entry
              </ModalHeader>,
              <ModalContent>
                <p>
                  Edit your entry below
                </p>
                <Form
                  as="form"
                  onSubmit={[Function]}
                >
                  <FormGroup>
                    <FormField
                      control={[Function]}
                      label="Date"
                      name="date"
                      onChange={[Function]}
                      placeholder="Enter the date here..."
                      required={true}
                      value="string"
                    />
                    <FormField
                      control={[Function]}
                      label="Entry"
                      name="text"
                      onChange={[Function]}
                      placeholder="Start typing your entry here..."
                      required={true}
                      value="string"
                    />
                  </FormGroup>
                </Form>
              </ModalContent>,
              <ModalActions>
                <Button
                  as="button"
                  negative={true}
                  onClick={[Function]}
                >
                  Cancel
                </Button>
                <Button
                  as="button"
                  content="Submit"
                  icon="checkmark"
                  labelPosition="right"
                  onClick={[Function]}
                  positive={true}
                />
              </ModalActions>,
            ],
            "closeOnDimmerClick": true,
            "closeOnDocumentClick": false,
            "closeOnEscape": undefined,
            "dimmer": true,
            "eventPool": "Modal",
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Edit Your Entry",
              },
              "ref": null,
              "rendered": "Edit Your Entry",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <p>
                    Edit your entry below
                  </p>,
                  <Form
                    as="form"
                    onSubmit={[Function]}
                  >
                    <FormGroup>
                      <FormField
                        control={[Function]}
                        label="Date"
                        name="date"
                        onChange={[Function]}
                        placeholder="Enter the date here..."
                        required={true}
                        value="string"
                      />
                      <FormField
                        control={[Function]}
                        label="Entry"
                        name="text"
                        onChange={[Function]}
                        placeholder="Start typing your entry here..."
                        required={true}
                        value="string"
                      />
                    </FormGroup>
                  </Form>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit your entry below",
                  },
                  "ref": null,
                  "rendered": "Edit your entry below",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "form",
                    "children": <FormGroup>
                      <FormField
                        control={[Function]}
                        label="Date"
                        name="date"
                        onChange={[Function]}
                        placeholder="Enter the date here..."
                        required={true}
                        value="string"
                      />
                      <FormField
                        control={[Function]}
                        label="Entry"
                        name="text"
                        onChange={[Function]}
                        placeholder="Start typing your entry here..."
                        required={true}
                        value="string"
                      />
                    </FormGroup>,
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <FormField
                          control={[Function]}
                          label="Date"
                          name="date"
                          onChange={[Function]}
                          placeholder="Enter the date here..."
                          required={true}
                          value="string"
                        />,
                        <FormField
                          control={[Function]}
                          label="Entry"
                          name="text"
                          onChange={[Function]}
                          placeholder="Start typing your entry here..."
                          required={true}
                          value="string"
                        />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "control": [Function],
                          "label": "Date",
                          "name": "date",
                          "onChange": [Function],
                          "placeholder": "Enter the date here...",
                          "required": true,
                          "value": "string",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "control": [Function],
                          "label": "Entry",
                          "name": "text",
                          "onChange": [Function],
                          "placeholder": "Start typing your entry here...",
                          "required": true,
                          "value": "string",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Button
                    as="button"
                    negative={true}
                    onClick={[Function]}
                  >
                    Cancel
                  </Button>,
                  <Button
                    as="button"
                    content="Submit"
                    icon="checkmark"
                    labelPosition="right"
                    onClick={[Function]}
                    positive={true}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "children": "Cancel",
                    "negative": true,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "content": "Submit",
                    "icon": "checkmark",
                    "labelPosition": "right",
                    "onClick": [Function],
                    "positive": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
